AEPIdentity/Tests/IdentityFunctionalTests.swift

```diff
@@ -467,4 +467,78 @@ class IdentityFunctionalTests: XCTestCase {
         let ecid = dispatchedEvent?.data?[IdentityConstants.EventDataKeys.VISITOR_ID_ECID] as? String ?? ""
         XCTAssertFalse(ecid.isEmpty)
     }
+    
+    // MARK: dispatch AnalyticsForIdentityRequest Event tests
+    
+    // Test setting valid push token on launch dispatches event to Analytics with a.push.optin = True
+    func testSetPushIdentifierValidTokenDispatchesAnalyticsForIdentityRequestEvent() {
+        // Set valid config to allow sync to process
+        identity.state?.lastValidConfig = [IdentityConstants.Configuration.GLOBAL_CONFIG_PRIVACY: PrivacyStatus.optedIn.rawValue]
+        
+        let encodedPushId = "746573742D707573682D6964" // encoded value of string "test-push-id"
+                
+        let data = ["pushidentifier": encodedPushId]
+        let event = Event(name: "Set Push Identifier", type: EventType.genericIdentity, source: EventSource.requestContent, data: data)
+        
+        mockRuntime.simulateComingEvent(event: event)
+        
+        let dispatchedEvent = mockRuntime.dispatchedEvents.first
+
+        XCTAssertEqual(EventType.analytics, dispatchedEvent?.type)
+        XCTAssertEqual(EventSource.requestContent, dispatchedEvent?.source)
+        XCTAssertEqual("Push", dispatchedEvent?.data?["action"] as? String)
+        XCTAssertTrue((dispatchedEvent?.data?["trackinternal"] as? Bool) ?? false)
+        let contextData = dispatchedEvent?.data?["contextdata"] as? [String: Any]
+        XCTAssertEqual("True", contextData?["a.push.optin"] as? String)
```

Comment:
- could use json comparison for easier reading
- can we check the persisted value for push token as well

Reply:
@kevinlind https://github.com/adobe/aepsdk-core-ios/pull/1002

We had to include the files as source because AEPTestUtils depends on Core and caused a circular dependency compilation error. We have a task for a long term solution in the future.